src/init.cpp

```diff
@@ -37,7 +38,7 @@ std::ostream& operator<<(std::ostream& os, const configuration& cfg) {
      << std::endl;
   os << "  umpire_device_memory_pool_initial_bytes = " << cfg.umpire_device_memory_pool_initial_bytes
      << std::endl;
-  os << "  mpi_pool = " << cfg.mpi_pool << std::endl;
+  os << "  mpi_pool = " << pika::mpi::experimental::get_pool_name() << std::endl;
```

Comment:
This doesn't hurt, but should we actually add the pool name also to pika's runtime configuration, so that `--pika:dump-config` prints the pool name, instead of leaving it up to DLA-Future to print it? We can also have both.

Reply:
Any comments on this?

Reply:
As DLAF has no settings for it but it comes form pika, IMO it is the best option.
On the other hand is pika dump-config invoked together with DLAF? Knowing if MPI pool was used is quite important.
Anyway, I would address this in another PR.